### PR TITLE
improve retcode checking

### PR DESCRIPTION
**Pull request by wizardofoz123**
In order to prepare skt for multirecipe/recipeset handling, we should
improve retcode checking.
This patch should mean no functional changes, but should make things
easier to read and manage.

Signed-off-by: Jakub Racek <jracek@redhat.com>

### CODE REVIEW

File: skt/runner.py [hunk 1]
```diff
@@ -28,6 +28,70 @@
...
+
+result_condition_checks = [
+    # This contains objects that will return <retval> (first parameter), when
+    # all the specified conditions are met. Empty conditions with no keywords
+    # are never met.
+
+    # Previous task was waived and panicked, which causes the next
+    # task to abort. The task is waived for a reason, return
+    # SKT_SUCCESS.
+    ConditionCheck(SKT_SUCCESS, result='Fail', waived=False, status='Aborted',
```
Comment (veruu) [inline skt/runner.py]: Shouldn't this be `result=Warn` ?

File: skt/runner.py [hunk 2]
```diff
@@ -291,12 +331,11 @@ def __getresults(self):
...
-                    if recipe_result.attrib.get('result') != 'Pass':
-                        ret = self.decide_run_result_by_task(recipe_result)
+                    ret = self.decide_run_result_by_task(recipe_result)
 
-                        if ret != SKT_SUCCESS:
-                            logging.info('Failure in a recipe detected!')
-                            return ret
+                    if ret != SKT_SUCCESS:
+                        logging.info('Failure in a recipe detected!')
+                        return ret
```
Comment (inakimalerba) [inline skt/runner.py]: This wouldn't change the issue we were discussing last friday on IRC about returning the first error that was found on the recipes.

Maybe errors should be saved and return the .. highest value? .

Comment (veruu) [reply skt/runner.py]: @inakimalerba it does not -- the PR just tries to simplify retcode checking here so the fix for the issue you mention is easier to manage

File: skt/runner.py (hunk 1 above)
Comment (wizardofoz123) [reply skt/runner.py]: I think so. The original code would actually allow warn/fail/panic in this check...

**Comment by wizardofoz123** [comment]
@cki-bot test

**Comment by wizardofoz123** [comment]
@cki-bot test

**Comment by wizardofoz123** [comment]
The pipelines are OK. Can we merge this @veruu ?

**Pull request merged**